dbms/src/Functions/array/arrayCompact.cpp

```diff
@@ -0,0 +1,110 @@
+#include <DataTypes/DataTypesNumber.h>
+#include <Columns/ColumnsNumber.h>
+#include "FunctionArrayMapped.h"
+#include <Functions/FunctionFactory.h>
+
+namespace DB
+{
+/// arrayCompact(['a', 'a', 'b', 'b', 'a']) = ['a', 'b', 'a'] - compact arrays
+    namespace ErrorCodes
+    {
+        extern const int ILLEGAL_COLUMN;
+    }
+
+    struct ArrayCompactImpl
+    {
+        static bool useDefaultImplementationForConstants() { return true; }
+        static bool needBoolean() { return false; }
+        static bool needExpression() { return false; }
+        static bool needOneArray() { return false; }
+
+        static DataTypePtr getReturnType(const DataTypePtr & expression_return, const DataTypePtr & /*array_element*/)
+        {
+            WhichDataType which(expression_return);
+
+            if (which.isUInt8()) { return std::make_shared<DataTypeArray>(std::make_shared<DataTypeUInt8>()); }
```

Comment:
To avoid writing out every numeric type, you can use `isNumber(which)`. Maybe you can even return Array of input type here and not check anything, because you'll check it in execute() anyway.

Reply:
Thanks. I have updated it.